### PR TITLE
Handle case where configuration change is being processed for a group that no longer exists.

### LINKED ISSUE #3010: raft: nil pointer dereference in multiNode.run
The following test appears to have failed:

[#8932](https://circleci.com/gh/cockroachdb/cockroach/8932):

```
I1105 00:35:09.234897 929 storage/replica_gc_queue.go:125  destroying local data from range 1
I1105 00:35:09.235841 929 storage/store.go:1475  changing raft replica {3 3 8} for range 1
E1105 00:35:09.237934 929 storage/store.go:1543  storage/store.go:1542: got committed raft command for 1 but have no range with that ID: {RangeID:1 OriginReplica:{NodeID:1 StoreID:1 ReplicaID:1} Cmd:BeginTransaction ["\x00\x00\x00k1\x00\x01rdsc",""), ConditionalPut ["\x00\x00\x00k1\x00\x01rdsc",""), Put ["\x00\x00meta2\xff\xff",""), Put ["\x00\x00meta1\xff\xff","")}
E1105 00:35:09.238132 929 storage/store.go:1543  storage/store.go:1542: got committed raft command for 1 but have no range with that ID: {RangeID:1 OriginReplica:{NodeID:1 StoreID:1 ReplicaID:1} Cmd:EndTransaction ["\x00\x00\x00k1\x00\x01rdsc","")}
W1105 00:35:09.239251 929 multiraft/multiraft.go:1151  aborting configuration change: storage/store.go:1542: got committed raft command for 1 but have no range with that ID: {RangeID:1 OriginReplica:{NodeID:1 StoreID:1 ReplicaID:1} Cmd:EndTransaction ["\x00\x00\x00k1\x00\x01rdsc","")}
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x8 pc=0xdc73ff]

goroutine 6856 [running]:
github.com/coreos/etcd/raft.(*multiNode).run(0xc820303c80)
	/go/src/github.com/coreos/etcd/raft/multinode.go:264 +0x174f
created by github.com/coreos/etcd/raft.StartMultiNode
	/go/src/github.com/coreos/etcd/raft/multinode.go:71 +0x338

goroutine 1 [chan receive]:
testing.RunTests(0x1891350, 0x1ea96a0, 0xc9, 0xc9, 0x1568201)
--
	/go/src/github.com/cockroachdb/cockroach/multiraft/storage.go:199 +0x1031
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820302840, 0xc82050fbc0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
FAIL	github.com/cockroachdb/cockroach/storage	19.019s
=== RUN   TestBatchBasics
I1105 00:34:48.318960 923 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I1105 00:34:48.323236 923 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I1105 00:34:48.329965 923 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

### PR DESCRIPTION
**Pull request by BramGruneir**
Fixes #3010
But exposes some other harder to reproduce bugs.  
I'll list them here before merging.  I've seen 3 so far.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3176)

### CODE REVIEW

File: multiraft/multiraft.go [hunk 1]
```diff
@@ -1154,7 +1154,12 @@ func (s *state) processCommittedEntry(groupID roachpb.RangeID, g *group, entry r
 			Callback: func(err error) {
 				select {
 				case s.callbackChan <- func() {
-					if err == nil {
+					gInner, ok := s.groups[groupID]
+					if !ok {
+						log.Infof("group %d no longer exists, aborting configuration change", groupID)
+					} else if gInner.groupID != groupID {
+						panic(fmt.Sprintf("group ids do not match, original:%d, current:%d", gInner.groupID, groupID))
```
Comment (bdarnell) [inline multiraft/multiraft.go]: The group IDs will always match; my suggestion was to assert that it's the same object (pointer equality: `gInner == g`).

**Comment by bdarnell** [comment]
LGTM. Also fixes #2950, right?

File: multiraft/multiraft.go (hunk 1 above)
Comment (BramGruneir) [inline multiraft/multiraft.go]: Fixed.

**Comment by BramGruneir** [comment]
Sadly, no, #2950 still happens.
I've seen a number of failure still (just harder to reproduce them that this one):

1. #3178
panic: requested entry at index is unavailable
github.com/coreos/etcd/raft.(*raftLog).term(0xc8203b6000, 0x8e, 0x100, 0x0, 0x0)
	/Users/bram/go/src/github.com/coreos/etcd/raft/log.go:231 +0x1a9

2. No issue yet
panic: [group 1] applied(160) is out of range [prevApplied(0), committed(0)]
github.com/cockroachdb/cockroach/multiraft.(*raftLogger).Panicf(0xc820594df0, 0x505f9e0, 0x3c, 0xc82067c270, 0x3, 0x3)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/multiraft/raft.go:121 +0x1a2

3. #2950 
panic: runtime error: invalid memory address or nil pointer dereference
github.com/coreos/etcd/raft.stepLeader(0xc8200d96c0, 0xb, 0x0, 0x16, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/coreos/etcd/raft/raft.go:590 +0x1649

**Pull request merged**